src/action.rs

```diff
@@ -16,14 +36,76 @@ pub enum HoldTapConfig {
     /// not used in the flow of typing, like escape for example. If
     /// you are annoyed by accidental tap, you can try this behavior.
     HoldOnOtherKeyPress,
-    /// If there is a release and a press of another key, the hold
+    /// If there is a press and release of another key, the hold
     /// action is activated.
     ///
     /// This behavior is interesting for fast typist: the different
     /// between hold and tap would more be based on the sequence of
     /// events than on timing. Be aware that doing the good succession
     /// of key might require some training.
     PermissiveHold,
+    /// A custom configuration. Allows the behavior to be controlled by a caller
+    /// supplied handler function.
+    ///
+    /// The input to the custom handler will be an iterator that returns
+    /// [Stacked] [Events](Event). The order of the events matches the order the
+    /// corresponding key was pressed/released, i.e. the first event is the
+    /// event first received after the HoldTap action key is pressed.
+    ///
+    /// The return value should be the intended action that should be used. A
+    /// [Some] value will cause one of: [WaitingAction::Tap] for the configured
+    /// tap action, [WaitingAction::Hold] for the hold action, and
+    /// [WaitingAction::NoOp] to force no action to occur this cycle. A [None]
+    /// value will cause a fallback to the timeout-based approach.
```

Comment:
Do you have an example when the difference between `None` and `Some(WaitingAction::NoOp)` would be interesting? Particularly, is there a case where not fallbacking to timeout is interesting?

Reply:
I could imagine a possible scenario where someone decides they don't want a key to activate if certain other keys are being pressed. I tend to err on the side of flexibility, but if you would prefer it without the Option that's fine too. It won't affect any of the use-cases I have for this feature.